src/H5FDsubfiling.c

```diff
@@ -433,18 +434,50 @@ H5FD__copy_plist(hid_t fapl_id, hid_t *id_out_ptr)
 herr_t
 H5Pset_fapl_subfiling(hid_t fapl_id, H5FD_subfiling_config_t *fa)
 {
-    H5P_genplist_t *plist     = NULL; /* Property list pointer */
-    herr_t          ret_value = FAIL;
+    H5P_genplist_t *  plist      = NULL; /* Property list pointer */
+    hid_t             ioc_fapl   = H5I_INVALID_HID;
+    H5FD_ioc_config_t ioc_config = {
+        0,
+    };
+    H5FD_subfiling_config_t subfiling_conf = {
+        0,
+    };
+    herr_t ret_value = FAIL;
 
     FUNC_ENTER_API(FAIL)
     H5TRACE2("e", "i*!", fapl_id, fa);
 
-    HDassert(fa != NULL);
+    if (NULL == (plist = H5P_object_verify(fapl_id, H5P_FILE_ACCESS)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a file access property list")
 
-    plist = H5P_object_verify(fapl_id, H5P_FILE_ACCESS);
+    if (fa == NULL) {
+        /* Create IOC fapl */
+        ioc_fapl = H5Pcreate(H5P_FILE_ACCESS);
+        if (H5I_INVALID_HID == ioc_fapl)
+            HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "can't create ioc fapl")
+
+        /* Get subfiling VFD defaults */
+        if (H5Pget_fapl_subfiling(fapl_id, &subfiling_conf) < 0)
+            HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "can't get subfiling fapl")
+
+        if (subfiling_conf.require_ioc) {
+            /* Get IOC VFD defaults */
+            if (H5Pget_fapl_ioc(ioc_fapl, &ioc_config) < 0)
+                HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "can't get ioc fapl")
```

Comment:
Since you use created the doc_fapl, I don't think you need to worry about pre-existing IOC fapl entries in this case.  However, calling fapl_get_ioc_defaults() with be a little quicker.

Reply:
I'll just remove it. Why does fapl_get_ioc_defaults have only one "_" and  fapl__get_subfiling_defaults have two in the name?

Reply:
Darned if I know.  Quincey was trying to require package private functions to use the double underscore (i.e. H5FD__), so it might have something to do with that.

Reply:
It does not work if I rm this.

Reply:
fapl_get_ioc_defaults is a static  function, I gues that is why it is one "_"

Reply:
Sorry, I didn't realize that fapl_get_ioc_defaults() was static, and thus not available outside of H5FDioc.  It is a very minor issue, and I wouldn't be surprised if the code changes so as to make the point moot.  I suggest you leave this as is.